src/picomsg/src/utils/functors.hpp

```diff
@@ -58,3 +58,12 @@ struct hash<pm_ipsec_ts> {
 };
 
 } /* namespace std { */
+
+template<typename T>
+struct c_obj_destroyer {
```

Comment:
This is actually more verbose to use than actually needed.
Why not turn it into a wrapper that calls `destroy` in its `dtor`?

Reply:
Added a wrapper